### PR TITLE
Method `Single::frequencies()` added

### PR DESCRIPTION
**Pull request by Smoren**
Hi @markrogoyski,

This is a draft of PR. If you like my implementation, I'll finish it.

BTW I think this method is more suitable for `Single` namespace, not for `Reduce` (as you suggest) because it returns a generator.

### CODE REVIEW

File: src/Single.php [hunk 1]
```diff
@@ -523,4 +525,42 @@ public static function skip(iterable $data, int $count, int $offset = 0): \Gener
...
+
+            $usages[$hash]++;
+            $totalCount++;
+        }
+
+        /**
+         * @var T $value
+         * @var int $usageCount
+         */
+        foreach (Multi::zip($values, $usages) as [$value, $usageCount]) {
```
Comment (markrogoyski) [inline src/Single.php]: Suggestion: Consider using `zipEqual` here instead, as these should be the same length. If for some reason they were not, best to expose the bug than silently let it cause errant behavior.

Comment (Smoren) [reply src/Single.php]: Done.

**Comment by Smoren** [comment]
Hi @markrogoyski, thank you, I used `zipEqual()` as you suggested.

Also I added `Stream::frequencies()` method and covered it with test.

BTW, I what do you think about a new namespace like `Stat` for this function? Or maybe use `Math`?

**Comment by markrogoyski** [comment]
Hi @Smoren,

Thank you for this pull request, and apologies at the delay at reviewing it.

I have a suggestion, let's create two functions, using the technical definitions:

- `frequencies` which will return the number of times an item occurs. See [frequency definition](https://www.merriam-webster.com/dictionary/frequency). 
- `relativeFrequencies` which will return the ratio of the frequency against the total count (Which is what you have implemented already). See [relative frequency defintion](https://www.merriam-webster.com/dictionary/relative%20frequency)

For reference, this is also the same terminology used in MathPHP.
```php
use MathPHP\Statistics\Distribution;

$grades = ['A', 'A', 'B', 'B', 'B', 'B', 'C', 'C', 'D', 'F'];

// Frequency distributions (frequency and relative frequency)
$frequencies          = Distribution::frequency($grades);         // [ A => 2,   B => 4,   C => 2,   D => 1,   F => 1   ]
$relative_frequencies = Distribution::relativeFrequency($grades); // [ A => 0.2, B => 0.4, C => 0.2, D => 0.1, F => 0.1 ]
```

Thoughts about this change?
Thanks,
Mark

**Comment by Smoren** [comment]
Hi @markrogoyski,

Your suggestion is implemented!

**Comment by markrogoyski** [comment]
Hi @Smoren,

Thanks for implementing the change. Looks good to me.

Mark

**Comment by Smoren** [comment]
Thank you @markrogoyski!

**Comment by markrogoyski** [comment]
Hi @Smoren,

Considering one of your earlier comments a bit more, I moved this functionality into the `Math` namespace. It seems a better fit there.

Thanks again,
Mark

**Comment by Smoren** [comment]
Thank you @markrogoyski!

**Pull request merged**